--- media_build_experimental/linux/Makefile.org	2012-03-03 10:24:15.141944623 +0100
+++ media_build_experimental/linux/Makefile	2012-03-03 10:24:55.411944643 +0100
@@ -44,8 +44,8 @@
 
 TODAY_TAR := linux-media-`date "+%Y-%m-%d"`.tar
 TODAY_TAR_MD5 := linux-media-`date "+%Y-%m-%d"`.tar.bz2.md5
-LATEST_TAR := http://linuxtv.org/downloads/drivers/linux-media-LATEST.tar.bz2
-LATEST_TAR_MD5 := http://linuxtv.org/downloads/drivers/linux-media-LATEST.tar.bz2.md5
+LATEST_TAR := http://linuxtv.org/downloads/drivers/linux-media-2012-02-29.tar.bz2 
+LATEST_TAR_MD5 := http://linuxtv.org/downloads/drivers/linux-media-2012-02-29.tar.bz2.md5 
 
 default:
 	make -C ..
