--- at-3.1.13.orig/debian/changelog
+++ at-3.1.13/debian/changelog
@@ -0,0 +1,604 @@
+at (3.1.13-1ubuntu1) precise; urgency=low
+
+  * Resynchronise with Debian.  Remaining changes:
+    - Replace init script with Upstart job, and bump build-dependency on
+      debhelper for Upstart-aware dh_installinit.
+    - Suggest an MTA rather than Recommending one.
+
+ -- Colin Watson <cjwatson@ubuntu.com>  Tue, 25 Oct 2011 15:37:19 +0100
+
+at (3.1.13-1) unstable; urgency=low
+
+  * New upstream release:
+    + Remove useless chdir(). (Closes: #587121)
+    + Create hardlink as priviledged user. (Closes: #597130)
+    + Add support for -r option required by POSIX. (Closes: #616697)
+    + Don't abort when job is removed just before we run it. (Closes: #309014)
+    + Do not strip binaries on install. (Closes: #436485)
+  * Recommend default-mta instead of exim4. (Closes: #542872)
+  * debian/control: Add dependency on ${misc:Depends}. (Closes: #586260)
+  * debian/control: Remove DM-Upload-Allowed.
+  * Update my email address.
+  * debian/source/format: Use source format 1.0.
+  * debian/copyright: Refer to specific versions of the GPL.
+  * Add debian/watch with a comment that Debian is upstream.
+  * debian/postinst: use "set -e" instead of "#!/bin/sh -e".
+  * debian/rules: Add build-arch, build-indep targets.
+  * Bump Standards-Version to 3.9.2.
+
+ -- Ansgar Burchardt <ansgar@debian.org>  Sat, 25 Jun 2011 15:05:40 +0200
+
+at (3.1.12-1ubuntu3) oneiric; urgency=low
+
+  * Rebuild with current debhelper to pick up the fix for LP #690640.
+
+ -- Colin Watson <cjwatson@ubuntu.com>  Mon, 16 May 2011 11:24:34 +0100
+
+at (3.1.12-1ubuntu2) maverick; urgency=low
+
+  * Perform queue hardlinks with full privileges (LP: #598824).
+
+ -- Kees Cook <kees@ubuntu.com>  Sun, 27 Jun 2010 12:13:43 -0700
+
+at (3.1.12-1ubuntu1) maverick; urgency=low
+
+  * Merge from Debian unstable (LP: #559117), remaining changes:
+    - Replace init script with Upstart job, and bump build-dependency on
+      debhelper for Upstart-aware dh_installinit
+    - debian/control: Add missing ${misc:Depends}
+    - Suggest: default-mta | mail-transport-agent instead of Recommends:
+      exim | mail-transport-agent
+  * Dropped changes:
+    - Set SENDMAIL to /usr/lib/sendmail: this is wrong, the correct path of
+      /usr/sbin/sendmail as mandated by policy for the m-t-a package is
+      already used in Debian.
+    - Depend on lsb >= 3.2-14: superseded by upstart job and merged in
+      Debian.
+    - Add the 'status' action to the init script: superseded by upstart job.
+
+ -- Steve Langasek <steve.langasek@ubuntu.com>  Thu, 17 Jun 2010 16:32:42 +0000
+
+at (3.1.12-1) unstable; urgency=low
+
+  * New upstream release:
+    + do not increment year twice (Closes: #404965) (LP: #312051)
+    + do not increment day of month twice (Closes: #386665)
+    + update documentation on removed environment variables
+      (Closes: #335374)
+    + add -f option to stay in foreground (Closes: #208367) - thanks to
+      Marcela Maslanova <mmaslano@redhat.com>
+    + error when day of month is 0 (Closes: #372671)
+    + make less assumptations on open file descriptors (Closes: #453707)
+    + add status action to init script (Closes: #506829)
+    + make stdout, stderr available to sendmail process (Closes: #316442)
+    + document -b option (LP: #156493)
+    + document 4-digit years and ISO date format (Closes: #359010)
+    + add test suite for date/time parser (Closes: #86662)
+      - add build-dep on perl (>= 5.10.1) for Test::More 0.92
+  * use dh_installinit (Closes: #406359)
+  * remove support for upgrading from pre-3.1.9
+  * remove rc, pam (moved upstream)
+  * remove Cyril Brulebois from Uploaders as requested on IRC
+  * now depends on lsb-base (>= 3.2-14) for status_of_proc in init script
+  * add dependency on libpam-runtime (>= 1.0.1-11) for
+    common-session-noninteractive
+  * update debian/copyright
+  * bump Standards-Version to 3.8.3 (no changes)
+  * debian/control: Add DM-Upload-Allowed: yes.
+
+ -- Ansgar Burchardt <ansgar@43-1.org>  Mon, 30 Nov 2009 20:03:34 +0900
+
+at (3.1.11-1ubuntu5) lucid; urgency=low
+
+  * rebuild rest of main for armel armv7/thumb2 optimization;
+    UbuntuSpec:mobile-lucid-arm-gcc-v7-thumb2
+
+ -- Alexander Sack <asac@ubuntu.com>  Fri, 05 Mar 2010 03:09:56 +0100
+
+at (3.1.11-1ubuntu4) karmic; urgency=low
+
+  FFE LP: #427356.
+
+  * Replace init script with Upstart job.
+  * debian/control:
+    - Add missing ${misc:Depends}
+    - Bump build-dependency on debhelper for Upstart-aware dh_installinit
+
+ -- Scott James Remnant <scott@ubuntu.com>  Tue, 15 Sep 2009 03:33:39 +0100
+
+at (3.1.11-1ubuntu3) karmic; urgency=low
+
+  * And drop the MTA back down from a Recommends to a Suggests, to not be
+    pulled in by default.
+
+ -- Steve Langasek <steve.langasek@ubuntu.com>  Fri, 21 Aug 2009 15:21:37 -0700
+
+at (3.1.11-1ubuntu2) karmic; urgency=low
+
+  * Restore the Recommends: default-mta | mail-transport-agent, which was
+    dropped in the latest merge.
+
+ -- Steve Langasek <steve.langasek@ubuntu.com>  Thu, 20 Aug 2009 15:20:31 +0000
+
+at (3.1.11-1ubuntu1) karmic; urgency=low
+
+  * Merge from debian unstable, remaining changes: LP: #415407
+    - debian/control: Depend on lsb >= 3.2-14, which has the
+      status_of_proc() function.
+    - debian/rc: Add the 'status' action (Debian Bug: #506829)
+    - Set SENDMAIL to /usr/lib/sendmail.
+
+ -- Bhavani Shankar <right2bhavi@gmail.com>  Tue, 18 Aug 2009 20:02:37 +0530
+
+at (3.1.11-1) unstable; urgency=low
+
+  * New "upstream" release:
+    + correct comment in atd.c (Closes: #372640)
+    + rc: do not set -e (Closes: #445489)
+    + panic.c: switch permissions before unlink (Closes: #518456, #538702)
+    + at.c: add -t option (Closes: #515262)
+    + at.c: exit successfully when -V is given (Closes: #292657)
+    + fix some typos (Closes: #491131)
+    + set version to 3.1.11 (Closes: #484515)
+
+  [ Cyril Brulebois ]
+  * Mark old changelog entries using “Old Changelog”.
+  * debian/rules:
+    + Stop setting SHELL to bash.
+    + Get rid of unneeded checkroot target.
+    + Get rid of unneeded local variables.
+    + Get rid of lines that are commented out.
+    + Get rid of obsolete source and diff targets.
+    + Get rid of unneeded dist and dsc targets.
+    + Get rid of unneeded variables.
+    + Use dh_install to install the config files.
+    + Call some dh_* tools at the end of the build.
+      - Call dh_md5sum (Closes: #309773)
+    + Call dh_install{changelogs,doc} to install docs.
+    + Use dh_installman/dh_compress to handle the manpages.
+    + Call dh_clean.
+    + Simplify calling configure.
+    + Use a build-stamp target.
+    + Use proper cleaning.
+    + Use “sed -i”, that's shorter.
+    + Get rid of the copyright, rules got rewritten.
+    + No longer make distclean errors.
+    + Remove additional removals, dh_clean will do that.
+    + Set SENDMAIL= for ./configure instead of using sed
+  * debian/control:
+    + Build-Depend on debhelper (>= 7).
+    + Update Maintainer/Uploaders according to ITA (Closes: #541013).
+    + Add Vcs-{Browser,Git} pointing to collab-maint/at.git
+  * Remove debian/conffiles, debhelper takes care of that.
+
+  [ Ansgar Burchardt ]
+  * debian/rules:
+    + use config.{guess,sub} from autotools-dev (Closes: #536294)
+    + call configure as documented in autotools-dev (Closes: #480825)
+    + call dh_strip
+    + remove duplicate ChangeLog
+  * debian/control:
+    + do no longer build-dep on ssmtp | mail-transport-agent
+      (Closes: #524187)
+    + downgrade dependency on mail-transport-agent to recommends
+      (Closes: #468326)
+    + downgrade Priority to standard
+    + bump Standards-Version to 3.8.2
+  * debian/prerm: do not ignore errors
+  * Copy updated debian/copyright from upstream
+
+ -- Ansgar Burchardt <ansgar@43-1.org>  Fri, 14 Aug 2009 20:24:03 +0200
+
+at (3.1.10.2ubuntu2) jaunty; urgency=low
+
+  * Set SENDMAIL to /usr/lib/sendmail in configure's env and drop ssmtp |
+    mail-transport-agent build-deps.
+
+ -- Loic Minier <lool@dooz.org>  Wed, 15 Apr 2009 12:22:47 +0200
+
+at (3.1.10.2ubuntu1) jaunty; urgency=low
+
+  * Merge from debian unstable, remaining changes:
+    - debian/control: Drop mail-transport-agent to a Suggest
+      and prefer real alternative postfix.
+    - debian/control: Depend on lsb >= 3.2-14, which has the
+      status_of_proc() function.
+    - debian/rc: Add the 'status' action (Debian Bug: #506829)
+
+ -- Dustin Kirkland <kirkland@ubuntu.com>  Mon, 24 Nov 2008 21:55:01 -0600
+
+at (3.1.10.2) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * Change Depends to include exim4 as alternative to mail-transport-agent.
+    (closes: #474999)
+  * Change Build-Depends to include ssmtp as alternative to
+    mail-transport-agent. This is what the Debian build daemons do anyway.
+    (closes: #491877)
+
+ -- Christoph Berg <myon@debian.org>  Tue, 23 Sep 2008 11:11:21 +0200
+
+at (3.1.10.1ubuntu3) intrepid; urgency=low
+
+  * debian/control: Depend on lsb >= 3.2-12ubuntu2, which has the
+    status_of_proc() function.
+  * debian/rc: Add the 'status' action (LP: #247091).
+
+ -- Dustin Kirkland <kirkland@ubuntu.com>  Wed, 09 Jul 2008 16:39:40 -0500
+
+at (3.1.10.1ubuntu2) intrepid; urgency=low
+
+  * debian/control: Drop mail-transport-agent to a Suggests: again, and prefer
+    real alternative postfix. (Accidentally dropped in the merge)
+
+ -- Martin Pitt <martin.pitt@ubuntu.com>  Fri, 16 May 2008 11:52:38 +0200
+
+at (3.1.10.1ubuntu1) intrepid; urgency=low
+
+  * Merge with Debian unstable. Remaining Ubuntu changes:
+    - debian/postinst: Do not install stop rc.d links for shutdown and reboot,
+      they are not needed.
+  * debian/rc: Remove 0 and 6 from Default-Stop: LSB header, too.
+
+ -- Martin Pitt <martin.pitt@ubuntu.com>  Thu, 15 May 2008 15:50:38 +0200
+
+at (3.1.10.1) unstable; urgency=low
+
+  * Non-maintainer uploaded to fix release goal issue.
+  * Correct LSB header of init.d script (Closes: #376780).
+
+ -- Petter Reinholdtsen <pere@debian.org>  Sun, 27 Jan 2008 06:15:14 +0100
+
+at (3.1.10) unstable; urgency=low
+
+  * Fix typo in init script (closes: #321141)
+  * Allow "at <time> - <number> minutes" to work (closes: #321325)
+  * Correct cannot find job logic for atrm (closes: #325253)
+  * Rewrite init script using lsb functions (closes: #335307)
+  * Correct help for atrm; -q is not supported (closes: #332417)
+
+ -- Ryan Murray <rmurray@debian.org>  Mon,  2 Jan 2006 23:08:51 -0800
+
+at (3.1.9) unstable; urgency=low
+
+  * Apply rewritten parser patch (closes: #70928)
+  * Several changes to be more SUSv3 compliant (closes: #133812)
+  * Remove yacc/flex generated files, and remove them in the clean target
+  * Remove /usr/doc handling from scripts (closes: #261467)
+  * use chown with : instead of . (closes: #205523)
+  * Add two precisions to format strings (closes: #220336)
+  * Fix manpage typos (closes: #153700, #175211)
+  * Use --retry option of start-stop-daemon (closes: #84351)
+  * Create /var/spool/cron/atjobs/.SEQ in the postinst (closes: #142810)
+  * Always rescan the queue after signals (closes: #194674)
+  * use 89 11 as parameters to update-rc.d defaults (closes: #200627)
+  * hold privs longer when creating pid file (closes: #13603)
+  * Add stdio.h for stderr to parsetime.y (closes: #130023)
+  * Use invoke-rc.d in maintainer scripts (closes: #162677)
+  * Allow usernames > 8 characters.  Don't write out usernames with at that
+    atd will later call corrupt (closes: #17561)
+  * Run as user daemon, rather than root (from 3.1.8-11ubuntu3) (closes: #251058)
+    + Derooted /usr/bin/at.
+    + at.c:
+      - Keep real uid, only switch to daemon gid when unlinking files (the
+        directory is daemon-group writeable, but the job files are owned by user).
+      - Instead of creating the job file and then fchown()'ing it (which does
+        not work when running as non-root), change to the real gid when
+        creating the file.
+    + atd.c:
+      - Don't check that the file gid is equal to the execution gid (since job
+        files are now always owned by "daemon".
+      - setgid() to the user's gid, not to the file gid for job execution.
+    + debian/rules:
+      - Install /usr/bin/at as daemon:daemon 6555 instead of root:root 4755
+        (running as user daemon is necessary to be allowed to send a signal to
+         atd.)
+      - Install /etc/at.deny as root:daemon 640 instead of root:root 0600.
+    + debian/postinst:
+      - Make /var/spool/cron/{atjobs,atspool} writeable for group "daemon".
+      - Update permissions of /etc/at.{allow,deny} if we upgrade from a previous
+        version and there is no statoverride for these files.
+  * Update standards version (closes: #239394)
+  * Increment tm_mday for at "time" case, rather than hardcoded increment of
+    a day in seconds.  (closes: #61604)
+  * Print usage on "at -c" (closes: #202617)
+  * Increment tm_year when the month and day are earlier than current
+    (closes: #79650)
+  * Drop ISO timeformat; following SUSv3 instead (closes: #53586)
+  * Add optional PAM support based on the submitted patch (closes: #68364)
+  * Use AC_FUNC_GETLOADAVG and getloadavg.c from XEmacs, so that the
+    system function is used first, then the getloadavg.c from at if it isn't
+    available (closes: #269440)
+  * Fix permissions of /var/spool/cron/at* in the deb files to match what is
+    used now
+  * Remove /usr/sbin/atrun; the packaging is set up for running as a daemon.
+  * Align batch(1) with SUSv3 by accepting no arguments (closes: #70988)
+  * Update description of -v option to clarify that it prints the time
+    before the job submission starts (closes: #257074)
+  * Update description of submission to uppercase queues (closes: #84791)
+  * Update atq format description (closes: #193396)
+  * Return EXIT_FAILURE if one jobid fails for -c and -r (closes: #173635)
+
+ -- Ryan Murray <rmurray@debian.org>  Tue,  2 Aug 2005 02:34:45 -0700
+
+at (3.1.8-11) unstable; urgency=high
+
+  * Temporarily taking over this package
+  * Incorporate security fix from stable
+  * Regenerate y.tab.* (closes: #129561)
+  * Apparently you can't find the so called "upstream" version anymore, and
+    other distributions list us as upstream, so making it Debian native
+    (closes: #129529)
+  * Fix typo in atd manpage (closes: #122615)
+  * Fix /usr/doc reference in at manpage (closes: #75262)
+  * Install prerm script (closes: #127450, #109949)
+  * Update build-depends (closes: #89824)
+  * Remove suidmanager support (closes: #61868, #74056)
+  * Build with -O2
+
+ -- Ryan Murray <rmurray@debian.org>  Fri, 18 Jan 2002 00:13:16 -0800
+
+at (3.1.8-10.1) stable; urgency=high
+
+  * Security upload
+  * Removed superflous free() which caused a segv
+  * Check for the return value of mktime()
+
+ -- Martin Schulze <joey@infodrom.org>  Sat, 12 Jan 2002 16:22:42 +0100
+
+at (3.1.8-10) frozen unstable; urgency=high
+
+  * Suidunregister /usr/bin (closes: Bug#59421).
+  * Rewrote perm.c to correctly deal with single character pw_name and
+    incomplete last lines in at.{allow,deny}. Thanks to
+    Sotiris.Vassilopoulos@betatech.gr for pointing this out.
+  * Included patch by Florian Zumbiehl <florz@gmx.de> adding some
+    read-only variables to no_export.
+  * SIGCHLD handler now correctly loops on a non-blocking waitpid.
+    Hopefully this gets rid of zombies even on SMP systems
+    (closes: Bug#60549).
+  * Changed AC_FUNC_GETLOADAVG to AC_REPLACE_FUNCS(getloadavg) in
+    configure.in since it introduces a spurious depencency on libelfg0
+    whenever that package is installed on the build machine.
+    (closes: Bug#61295). 
+  * configure.in: rewrote directory tests such that configure fails due to
+    a missing SPOOLDIR only when --with-jobdir or --with-atspool doesn't
+    specify an explicit path.
+
+ -- Siggy Brentrup <bsb@winnegan.de>  Mon,  3 Apr 2000 13:56:47 +0200
+
+Old Changelog:
+
+at (3.1.8-9) frozen unstable; urgency=low, closes=53715 56047 56607 55560 55514
+
+  * Added SIGCHLD handler to release zombies (closes 53715 56047 56607)
+  * Changed dependency to mail-transport-agent only since /usr/bin/mail
+    doesn't grok the Subject: line. (closes 55560)
+  * Fixed postinst to suidregister /usr/bin/at (closes 55514)
+
+ -- Siggy Brentrup <bsb@winnegan.de>  Sun, 30 Jan 2000 22:00:46 +0100
+
+at (3.1.8-8) unstable; urgency=low
+
+  * New maintainer (according to Joey this is an upstream job too :)
+  * Upgrade to standards version 3.1.1.1
+
+ -- Siggy Brentrup <bsb@winnegan.de>  Tue, 11 Jan 2000 10:08:06 +0100
+
+at (3.1.8-7.1) unstable; urgency=low
+
+  * Non-maintainer release, with permission.
+  * Corrected patch for Bug#39527 so that it doesn't ignore all pending jobs.
+    (closes: #49708, #49739, #50984).
+
+ -- Richard Braakman <dark@xs4all.nl>  Thu, 25 Nov 1999 23:32:43 +0100
+
+at (3.1.8-7) unstable; urgency=low, closes=38293 25070 39527 38293 43032 38962 42469
+
+  * Changed load average to run to 1.5 (closes: Bug#38293)
+  * Unignored SIGCHLD (closes: Bug#25070)
+  * Applied patch from Zack Weinberg <zack@rabi.columbia.edu> that keeps
+    atd from spinning the disk (closes: Bug#39527)
+  * Added support for arguments to /etc/init.d/atd (closes: Bug#38293)
+  * Reworked the date parser so it will recognize 2345 and 102900
+    properly, 12900 as well (closes: Bug#43032, Bug#38962)
+  * Corrected weekday parser (closes: Bug#42469)
+  * Corrected path in atrun(8).
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Thu, 28 Oct 1999 00:05:52 +0200
+
+at (3.1.8-6) unstable; urgency=low, closes=34919 36628 36650
+
+  * Added text -x atd to init.d file (closes: Bug#34919)
+  * Added parsetest target to Makefile.in to provide a test program
+  * Hax0red the date/time parser so it parses the date properly,
+    especially when specifying a year (closes: Bug#36628)
+  * Added -M (don't send mail at all) feature to at client and at server
+    (closes: Bug#36650)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Sun, 23 May 1999 18:15:33 +0200
+
+at (3.1.8-5) unstable; urgency=low, closes=30796 29843
+
+  * Added proper support for time specified with AM/pm.  Thanks to Julian
+    Gilbey <jdg@maths.qmw.ac.uk> (closes: Bug#30796)
+  * Added more system accounts to at.deny file (closes: Bug#29843)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Sun,  3 Jan 1999 17:02:56 +0100
+
+at (3.1.8-4) unstable; urgency=low, closes=27414 19272 24943 11633 17131 27872 26917 25227 23141 21528 18181
+
+  * Applied OpenBSD/buflen patch provided by Topi Miettinen
+    <tom@medialab.sonera.net> (closes: Bug#27414)
+  * Corrected at_allow(5) manpage (closes: Bug#24943)
+  * Moved /bin/sh warning in front of input stream (closes: Bug#11633)
+  * Added prototype for yylex() and yyerror() (closes: Bug#17131)
+  * Corrected recognition of weekdays (closes: Bug#27872, Bug#26917, Bug#18181)
+  * Reformatted some "SEE ALSO" section.
+  * Adjusted Usage output of atq according to code and manpage.  `-v' is
+    not supported anymore. (closes: Bug#25227, Bug#21528)
+  * Re-added support for atq displaying the owner of the atjob.  I dunno
+    if the author would accept it if he would appear again, though.
+    (closes: Bug#23141)
+  * Removed preinst
+  * Documented problem with NFS-mounted spool directories (Bug#13603 and
+    Bug#13605)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Sat, 24 Oct 1998 12:59:08 +0200
+
+at (3.1.8-3) unstable; urgency=low, closes=19272 14529 15812 16429 16581 25981 25982
+
+  * New maintainer
+  * Updated Standards-Version
+  * Rewrote rules file
+  * Fixed manpages
+  * Renamed copyright (closes: Bug#14529, fixes: Bug#16429)
+  * Fixed misspelled files in /usr/doc/at (fixes: Bug#16429)
+  * Adjusted copyright file
+  * Added /etc/init.d/atd as conffile
+  * Completed init.d script
+  * Aded --with-jobdir and --with-atspool so it builds on systems where it
+    isn't installed as well (closes: Bug#15812)
+  * Corrected location of sendmail (closes: Bug#16429)
+  * Fixed preinst (closes: Bug#16581)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Fri, 23 Oct 1998 20:54:46 +0200
+
+at (3.1.8-2.1.1) unstable; urgency=low
+
+  * Non maintainer upload.
+    config.{guess,sub} changed to recognize a Arm architecture.
+
+ -- Turbo Fredriksson <turbo@debian.org>  Thu, 20 Aug 1998 04:48:23 -0400
+
+at (3.1.8-2.1) frozen unstable; urgency=low
+
+  * Non-maintainer upload.
+  * Provides correct name of copyright file, fixing important bug
+    #19272
+
+ -- John Goerzen <jgoerzen@complete.org>  Sat, 18 Apr 1998 23:59:28 -0500
+
+at (3.1.8-2) unstable; urgency=medium
+  * libc6 version
+  * (as in -1) Fix date parsing bug with 'next monday' where the current day
+  *   is Mondy
+  * Fix date parsing bug with February
+  * Misc. manpage changes
+ -- Thomas Koenig <ig25@rz.uni-karlsruhe.de>  Mon, 29 Sep 1997 17:21:40 +0200
+
+at (3.1.8-1) stable; urgency=low
+  * ../at_3.1.7-6_i386.deb
+  *
+  * libc5 version
+  * Fix date parsing bug with 'next monday' where the current day is monday
+  * Fix date parsing bug with February
+  * Misc. manpage changes
+ -- Thomas Koenig <ig25@rz.uni-karlsruhe.de>  Sun, 28 Sep 1997 21:33:44 +0200
+   
+at (3.1.7-6) unstable; urgency=high
+  * ../at_3.1.7-4_i386.deb
+  *
+  * libc6 version
+  * should fix installation bugs
+ -- Thomas Koenig <ig25@rz.uni-karlsruhe.de>  Mon, 22 Sep 1997 20:26:05 +0200
+
+at (3.1.7-5) stable; urgency=low
+  * ../at_3.1.7-5_i386.deb
+  *
+  * libc5 version
+  * Make runlevel detection more robust
+ -- Thomas Koenig <ig25@rz.uni-karlsruhe.de>  Mon, 22 Sep 1997 15:48:55 +0200
+
+at (3.1.7-4) unstable; urgency=high
+  * ../at_3.1.7-4_i386.deb
+  *
+  * libc6 version
+  * removes spurious dependency on libelf
+  * Start atd if runlevel cannot be determined
+ -- Thomas Koenig <ig25@rz.uni-karlsruhe.de>  Tue, 15 Sep 1997 14:16:24 +0200
+
+at (3.1.7-3) stable; urgency=low
+  * ../at_3.1.7-3_i386.deb
+  * 
+  * libc5 version
+  * removed spurious dependendy on libelf0
+ -- Thomas Koenig <ig25@rz.uni-karlsruhe.de>  Tue, 17 Jun 1997 18:01:33 +0200
+
+at (3.1.7-2) unstable; urgency=low
+  * ../at_3.1.7-2_i386.deb
+  * Recompiled for libc6 (hamm)
+ -- Thomas Koenig <ig25@rz.uni-karlsruhe.de>  Sat, 14 Jun 1997 21:01:17 +0200
+
+at (3.1.7-1) unstable frozen; urgency=medium
+  * ../at_3.1.7-1_i386.deb
+  * Make upcase words work
+  * Start atd if runlevel can't be determined
+ -- Thomas Koenig <ig25@rz.uni-karlsruhe.de>  Mon, 26 May 1997 10:38:06 +0200
+
+at (3.1.6-1) unstable frozen; urgency=medium
+  * ../at_3.1.6-1_i386.deb
+  * Fix broken "noon", "teatime" and "midnight"
+ -- Thomas Koenig <ig25@rz.uni-karlsruhe.de>  Tue, 13 May 1997 14:50:40 +0200
+
+at (3.1.5-2) unstable frozen; urgency=high
+
+  * 
+  * ../at_3.1.5-2_i386.deb
+  * Also check for /etc/suid.conf when calling suidregister
+ -- Thomas Koenig <ig25@rz.uni-karlsruhe.de>  Wed, 7 May 1997 11:14:49 +0200
+
+at (3.1.5-1) unstable frozen; urgency=high
+
+  * ../at_3.1.5-1_i386.deb
+  * New "upstream" release:  Fixes bug #9390
+  * Removed "set -e" from prerm in an effort to fix #9311
+  * From now on, changelog will be more canonical (#9236)
+ -- Thomas Koenig <ig25@rz.uni-karlsruhe.de>  Tue, 6 May 1997 19:50:02 +0200
+
+at (3.1.4-2) unstable frozen; urgency=low
+
+  * ../at_3.1.4-2_i386.deb
+  * Start atd in postinst only if runlevel is right (bug #9073)
+  * Fixed extra space in /etc/init.d/atd (bug #9046)
+
+at (3.1.4-1) unstable frozen; urgency=low
+
+  * ../at_3.1.4-1_i386.deb
+  * New "upstream" release
+
+at (3.1.3-1) unstable frozen; urgency=low
+
+  * ../at_3.1.3-2_i386.deb
+  * make distclean on debian/rules clean
+  * depend on mail-transport-agent or mailx
+
+at (3.1.3-1) unstable frozen; urgency=low
+
+  * ../at_3.1.3-1_i386.deb
+  * New "upstream release", bug #8716 fixed
+
+at (3.1.2-2) unstable; urgency=low
+
+  * ../at_3.1.2-2_i386.deb
+  * adapted aj's /etc/init.d/atd (bug #8501)
+  * Fix bug #8495 (exit preinst cleanly when no /etc/crontab)
+  * Fix bug #7936 (for real this time)
+
+at (3.1.2-1) unstable; urgency=low
+
+  * ../at_3.1.2-1_i386.deb
+  * Fixed bug #8189 (prerm/postrm failure).
+  * New "upstream" version.
+
+at (3.1.1-1) unstable; urgency=low
+
+  * ../at_3.1-1_i386.deb
+  * Fixed bug #7936 and #7808
+
+at (3.0-1) unstable; urgency=low
+
+  * ../at_3.0-1_i386.deb
+  * Initial Release.
+
+ -- Thomas Koenig <ig25@rz.uni-karlsruhe.de>  Thu, 20 Feb 1997 17:33:12 +0100
--- at-3.1.13.orig/debian/rules
+++ at-3.1.13/debian/rules
@@ -0,0 +1,77 @@
+#!/usr/bin/make -f
+
+confflags = --prefix=/usr --with-loadavg_mx=1.5 \
+            --with-jobdir=/var/spool/cron/atjobs \
+            --with-atspool=/var/spool/cron/atspool \
+            SENDMAIL=/usr/sbin/sendmail
+
+export DEB_HOST_GNU_TYPE  ?= $(shell dpkg-architecture -qDEB_HOST_GNU_TYPE)
+export DEB_BUILD_GNU_TYPE ?= $(shell dpkg-architecture -qDEB_BUILD_GNU_TYPE)
+
+ifeq ($(DEB_BUILD_GNU_TYPE), $(DEB_HOST_GNU_TYPE))
+  confflags += --build $(DEB_HOST_GNU_TYPE)
+else
+  confflags += --build $(DEB_BUILD_GNU_TYPE) --host $(DEB_HOST_GNU_TYPE)
+endif
+
+build: build-arch build-indep
+build-arch: build-stamp
+build-stamp:
+	[ -f config.sub.ori ] || mv config.sub config.sub.ori
+	[ -f config.guess.ori ] || mv config.guess config.guess.ori
+	ln -sf /usr/share/misc/config.sub .
+	ln -sf /usr/share/misc/config.guess .
+
+	./configure $(confflags)
+	$(MAKE)
+	$(MAKE) test
+	touch $@
+
+build-indep:
+# Nothing to be done here.
+
+clean:
+	[ ! -f Makefile ] || $(MAKE) distclean
+	[ ! -f config.sub.ori ] || mv -f config.sub.ori config.sub
+	[ ! -f config.guess.ori ] || mv -f config.guess.ori config.guess
+	dh_clean
+
+binary-indep:
+# Nothing to be done here
+
+binary-arch: build-stamp
+	dh_testroot
+	dh_prep
+	# Install docs:
+	dh_installchangelogs ChangeLog
+	dh_installdocs README
+	# Install everything:
+	$(MAKE) IROOT=$(CURDIR)/debian/at mandir=/usr/share/man docdir=/usr/share/doc install
+	# Remove (duplicate) changelog and copyright:
+	rm debian/at/usr/share/doc/at/Copyright
+	rm debian/at/usr/share/doc/at/ChangeLog
+	# TODO: Take care of those:
+	rm -f debian/at/var/spool/cron/atjobs/.SEQ
+	chown daemon:daemon debian/at/var/spool/cron/atjobs debian/at/var/spool/cron/atspool
+	# TODO: Why removing those?
+	rm debian/at/usr/sbin/atrun
+	rm debian/at/usr/share/man/man8/atrun.8
+	# Install/rename pam config file:
+	dh_install pam.conf etc/pam.d
+	mv debian/at/etc/pam.d/pam.conf \
+		debian/at/etc/pam.d/atd
+	# Install/rename init script:
+	dh_installinit --name=atd -u"start 89 2 3 4 5 . stop 11 1 ."
+	# Needed to keep in line with the previous upload:
+	chown root:daemon debian/at/etc/at.deny
+	dh_compress
+	dh_installdeb
+	dh_shlibdeps
+	dh_strip
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+binary: binary-indep binary-arch
+
+.PHONY: binary binary-arch binary-indep clean
--- at-3.1.13.orig/debian/watch
+++ at-3.1.13/debian/watch
@@ -0,0 +1 @@
+# Debian is upstream for this package.
--- at-3.1.13.orig/debian/at.atd.upstart
+++ at-3.1.13/debian/at.atd.upstart
@@ -0,0 +1,14 @@
+# atd - deferred execution scheduler
+#
+# at is a standard UNIX program that runs user-specified programs at
+# scheduled deferred times
+
+description	"deferred execution scheduler"
+
+start on runlevel [2345]
+stop on runlevel [!2345]
+
+expect fork
+respawn
+
+exec atd
--- at-3.1.13.orig/debian/compat
+++ at-3.1.13/debian/compat
@@ -0,0 +1 @@
+7
--- at-3.1.13.orig/debian/control
+++ at-3.1.13/debian/control
@@ -0,0 +1,24 @@
+Source: at
+Section: admin
+Priority: standard
+Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
+XSBC-Original-Maintainer: Ansgar Burchardt <ansgar@debian.org>
+Standards-Version: 3.9.2
+Build-Depends: debhelper (>= 7.3.15ubuntu2), autotools-dev, bison, flex, libpam0g-dev,
+ perl (>= 5.10.1)
+Vcs-Browser: http://git.debian.org/?p=collab-maint/at.git
+Vcs-Git: git://git.debian.org/git/collab-maint/at.git
+
+Package: at
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends},
+ lsb-base (>= 3.2-14), libpam-runtime (>= 1.0.1-11)
+Suggests: default-mta | mail-transport-agent
+Description: Delayed job execution and batch processing
+ At and batch read shell commands from standard input
+ storing them as a job to be scheduled for execution in the
+ future.
+ .
+ Use
+  at    to run the job at a specified time
+  batch to run the job when system load levels permit
--- at-3.1.13.orig/debian/copyright
+++ at-3.1.13/debian/copyright
@@ -0,0 +1,64 @@
+This is the Debian GNU/Linux prepackaged version of the deferred
+execution scheduler called at.
+
+This package was debianized by its author Thomas Koenig
+<ig25@rz.uni-karlsruhe.de>, taken over and re-packaged first by Martin
+Schulze <joey@debian.org> and then by Siggy Brentrup <bsb@winnegan.de>,
+and then taken over by Ryan Murray <rmurray@debian.org>.
+
+In August 2009 the upstream development and Debian packaging were taken over
+by Ansgar Burchardt <ansgar@debian.org> and Cyril Brulebois <kibi@debian.org>.
+
+This may be considered the experimental upstream source, and since there
+doesn't seem to be any other upstream source, the only upstream source.
+
+   Copyright: 1993, 1994, 1995, 1996, 1997 (c) Thomas Koenig
+              1993 (c) David Parsons
+	      2002, 2005 (c) Ryan Murray
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+On Debian systems, the complete text of version 2 of the GNU General
+Public License can be found in `/usr/share/common-licenses/GPL-2'.
+
+The files posixtm.c and posixtm.h have the following copyright information:
+
+   Copyright (C) 1989, 1990, 1991, 1998, 2000, 2001, 2002, 2003, 2004,
+   2005, 2006, 2007 Free Software Foundation Inc.
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+On Debian systems, the complete text of version 3 of the GNU General
+Public License can be found in `/usr/share/common-licenses/GPL-3'.
+
+parsetime.pl:
+
+  Copyright (C) 2009, Ansgar Burchardt <ansgar@debian.org>
+
+  Permission to use, copy, modify, and/or distribute this software for any
+  purpose with or without fee is hereby granted, provided that the above
+  copyright notice and this permission notice appear in all copies.
+
+  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+  REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+  AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+  LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
+  OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+  PERFORMANCE OF THIS SOFTWARE.
--- at-3.1.13.orig/debian/postinst
+++ at-3.1.13/debian/postinst
@@ -0,0 +1,13 @@
+#!/bin/sh
+set -e
+
+if [ "$1" = "configure" ]; then
+    SEQFILE=/var/spool/cron/atjobs/.SEQ
+    if [ ! -f $SEQFILE ]; then
+	echo 0 > $SEQFILE
+	chmod 600 $SEQFILE
+	chown daemon:daemon $SEQFILE
+    fi
+fi
+
+#DEBHELPER#
--- at-3.1.13.orig/debian/postrm
+++ at-3.1.13/debian/postrm
@@ -0,0 +1,8 @@
+#!/bin/sh
+set -e
+if [ "$1" = purge ]
+then
+	rm -rf /var/spool/cron/atjobs /var/spool/cron/atspool
+fi
+
+#DEBHELPER#
--- at-3.1.13.orig/debian/source/format
+++ at-3.1.13/debian/source/format
@@ -0,0 +1 @@
+1.0
